### ISSUE TITLE
Event `open` not being triggered when connecting to replica set with option `connectWithNoPrimary` (>= 4.8)

### ISSUE THREAD

**Issue by hbarcelos** [Automattic/mongoose#5264]
I'm trying to connect to a remote replica set using `mongoose@4.9.9`.

Here is the test script:

```js
const connectionString  = 'mongodb://host1,host2,host3/database?authSource=admin'
const options = {
  user,
  pass : password,
  db : {
    native_parser : false
  },
  replset : {
    ssl                  : true,
    replicaSet : 'rs',
    connectWithNoPrimary : true
  }
}

const connection = mongoose.createConnection(connectionString, options)

connection.on('open', () => {
  console.log('Connection open!')
})

connection.on('fullsetup', () => {
  console.log('Connected with full setup!')
})

connection.on('all', () => {
  console.log('Connected to all instances!')
})

connection.on('error', (err) => {
  console.log('Connection error!')
  console.trace(err)
})
```

The initial problem was that the `'open'` event was never triggered, even though the `'fullsetup'` and `'all'` were.

Then I decided to check other versions. I started with `4.4.x` and tried one minor version upgrade at a time.

Until `4.7.9`, the script worked as expected. The output is:
> Connection open!
> Connected with full setup!
> Connected to all instances!

However, as of `4.8.0`, the output is:
> Connected with full setup!
> Connected to all instances!

Then I tried to remove the `connectWithNoPrimary` option. This way, all versions until `4.9.9` behave correctly:
> Connection open!
> Connected with full setup!
> Connected to all instances!

**Comment by vkarpov15** [comment]
Thanks for reporting, will investigate

**Comment by vkarpov15** [comment]
Are you sure you can connect to all members of your replica set? Also, can you try upgrading to 4.10.2? Below script works fine with 4.10.2 and a replica set created using this [script](https://github.com/Automattic/mongoose/blob/master/tools/repl.js):

```javascript
const mongoose = require('mongoose');

const connectionString  = 'mongodb://localhost:31000,localhost:31001,localhost:31002/database'
const options = {
  db : {
    native_parser : false
  },
  replset : {
    replicaSet : 'rs',
    connectWithNoPrimary: true
  }
}

const connection = mongoose.createConnection(connectionString, options)

connection.on('open', () => {
  console.log('Connection open!')
})

connection.on('fullsetup', () => {
  console.log('Connected with full setup!')
})

connection.on('all', () => {
  console.log('Connected to all instances!')
})

connection.on('error', (err) => {
  console.log('Connection error!')
  console.trace(err)
})
```

**Comment by hbarcelos** [comment]
> Are you sure you can connect to all members of your replica set?

Absolutely sure. However, I'm trying to connect to remote hosts, not local.

The latest version available at the time I tested was `4.9.9`. I'll try with `4.10.2` and come back to you.

**Comment by varunjayaraman** [comment]
@hbarcelos did you try this with `4.10.2`?
